src/Backend/OnboardingInsuranceAPI/Areas/User/RegisterUser.cs

```diff
@@ -6,29 +6,31 @@
 
 namespace OnboardingInsuranceAPI.Areas.User;
 
-public class RegisterUserHandler : IHandler
+public class RegisterUser : IHandler
 {
     private readonly DataContext _context;
-    private readonly ILogger<RegisterUserHandler> _log;
+    private readonly ILogger<RegisterUser> _log;
 
-    public RegisterUserHandler(DataContext context, ILogger<RegisterUserHandler> log)
+    public RegisterUser(DataContext context, ILogger<RegisterUser> log)
     {
         _context = context;
         _log = log;
     }
 
     public async Task CreateUser(string pid, string email)
     {
-        var user = await _context.Users.FirstOrDefaultAsync(x => x.Pid == pid);//check if user exist
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Pid == pid);//check if user exist
         if (user == null)
```

Comment:
Revert the check. It is usually more cleaner to check for negatives rather then positives. Q.E.D: if user is not null return that user already exists. Also you don't say to user what happened, if you try to create user that exists this would probably just return 200. That is not acceptable.

Reply:
Ok

Reply:
This handler runs every sign-up, sign-in, checking if user exist, so its registered user, otherwise its add user into cosmos db to complete registration. On custom policies there is only one way I found, i didn't find how to call function only when sign-up, so its checking every time if user registered or not. And its becomes as feature, because registration can be not completed on Cosmos DB side,  so its checking if registration is ok every sign-in.